File: src/pt/joaomneto/ffgbutil/adventure/impl/SOBAdventure.java
package pt.joaomneto.ffgbutil.adventure.impl;

import java.io.BufferedWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import pt.joaomneto.ffgbutil.R;
import pt.joaomneto.ffgbutil.adventure.Adventure;
import android.os.Bundle;
import android.view.Menu;

public class SOBAdventure extends Adventure {

	private int currentCrewStrength = -1;
	private int currentCrewStrike = -1;
	private int initialCrewStrength = -1;
	private int initialCrewStrike = -1;
	private int log = -1;
	private List<String> booty = new ArrayList<String>();

	protected static final int FRAGMENT_SHIP_COMBAT = 2;
	protected static final int FRAGMENT_BOOTY = 4;
	protected static final int FRAGMENT_NOTES = 5;

	public SOBAdventure() {
		super();
		fragmentConfiguration.clear();
		fragmentConfiguration.put(FRAGMENT_VITAL_STATS, new AdventureFragmentRunner(R.string.vitalStats, "pt.joaomneto.ffgbutil.adventure.impl.fragments.sob.SOBAdventureVitalStatsFragment"));
		fragmentConfiguration.put(FRAGMENT_COMBAT, new AdventureFragmentRunner(R.string.fights, "pt.joaomneto.ffgbutil.adventure.impl.fragments.AdventureCombatFragment"));
		fragmentConfiguration.put(FRAGMENT_SHIP_COMBAT, new AdventureFragmentRunner(R.string.shipCombat, "pt.joaomneto.ffgbutil.adventure.impl.fragments.sob.SOBShipCombatFragment"));
		fragmentConfiguration.put(FRAGMENT_EQUIPMENT, new AdventureFragmentRunner(R.string.goldEquipment, "pt.joaomneto.ffgbutil.adventure.impl.fragments.AdventureEquipmentFragment"));
		fragmentConfiguration.put(FRAGMENT_BOOTY, new AdventureFragmentRunner(R.string.booty, "pt.joaomneto.ffgbutil.adventure.impl.fragments.sob.SOBAdventureBootyFragment"));
		fragmentConfiguration.put(FRAGMENT_NOTES, new AdventureFragmentRunner(R.string.notes, "pt.joaomneto.ffgbutil.adventure.impl.fragments.AdventureNotesFragment"));
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		try {
			super.onCreate(savedInstanceState);

		} catch (Exception e) {
			e.printStackTrace();
		}

	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		getMenuInflater().inflate(R.menu.adventure, menu);
		return true;
	}

	@Override
	public void storeAdventureSpecificValuesInFile(BufferedWriter bw) throws IOException {

		bw.write("currentCrewStrength="+currentCrewStrength+"\n");
		bw.write("currentCrewStrike="+currentCrewStrike+"\n");
		bw.write("initialCrewStrength="+initialCrewStrength+"\n");
		bw.write("initialCrewStrike="+initialCrewStrike+"\n");
		bw.write("log=" + getLog() + "\n");
		bw.write("gold=" + getGold() + "\n");
		bw.write("booty="+arrayToString(getBooty())+"\n");
	}

	@Override
	protected void loadAdventureSpecificValuesFromFile() {

		setCurrentCrewStrength(Integer.valueOf(getSavedGame().getProperty("currentCrewStrength")));
		setCurrentCrewStrike(Integer.valueOf(getSavedGame().getProperty("currentCrewStrike")));
		setInitialCrewStrength(Integer.valueOf(getSavedGame().getProperty("initialCrewStrength")));
		setInitialCrewStrike(Integer.valueOf(getSavedGame().getProperty("initialCrewStrike")));
		setLog(Integer.valueOf(getSavedGame().getProperty("log")));
		setGold(Integer.valueOf(getSavedGame().getProperty("gold")));
		setBooty(stringToArray(getSavedGame().getProperty("booty")));
	}

	public int getCurrentCrewStrength() {
		return currentCrewStrength;
	}

	public void setCurrentCrewStrength(int currentCrewStrength) {
		this.currentCrewStrength = currentCrewStrength;
	}

	public int getCurrentCrewStrike() {
		return currentCrewStrike;
	}

	public void setCurrentCrewStrike(int currentCrewStrike) {
		this.currentCrewStrike = currentCrewStrike;
	}

	public int getInitialCrewStrength() {
		return initialCrewStrength;
	}

	public void setInitialCrewStrength(int initialCrewStrength) {
		this.initialCrewStrength = initialCrewStrength;
	}

	public int getInitialCrewStrike() {
		return initialCrewStrike;
	}

	public void setInitialCrewStrike(int initialCrewStrike) {
		this.initialCrewStrike = initialCrewStrike;
	}

	public int getLog() {
		return log;
	}

	public void setLog(int log) {
		this.log = log;
	}

	public List<String> getBooty() {
		return booty;
	}

	public void setBooty(List<String> booty) {
		this.booty = booty;
	}

	

}

File: src/main/java/pt/joaomneto/titancompanion/adventure/impl/SOBAdventure.kt
package pt.joaomneto.titancompanion.adventure.impl

import android.os.Bundle
import android.view.Menu
import pt.joaomneto.titancompanion.R
import pt.joaomneto.titancompanion.adventure.Adventure
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureCombatFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureEquipmentFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureNotesFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.sob.SOBAdventureBootyFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.sob.SOBAdventureVitalStatsFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.sob.SOBShipCombatFragment
import pt.joaomneto.titancompanion.util.AdventureFragmentRunner
import java.io.BufferedWriter
import java.io.IOException
import java.util.ArrayList

class SOBAdventure : Adventure(
    arrayOf(
        AdventureFragmentRunner(R.string.vitalStats, SOBAdventureVitalStatsFragment::class),
        AdventureFragmentRunner(R.string.fights, AdventureCombatFragment::class),
        AdventureFragmentRunner(R.string.shipCombat, SOBShipCombatFragment::class),
        AdventureFragmentRunner(R.string.goldEquipment, AdventureEquipmentFragment::class),
        AdventureFragmentRunner(R.string.booty, SOBAdventureBootyFragment::class),
        AdventureFragmentRunner(R.string.notes, AdventureNotesFragment::class)
    )
) {

    var currentCrewStrength = -1
    var currentCrewStrike = -1
    var initialCrewStrength = -1
    var initialCrewStrike = -1
    var log = -1
    var booty: List<String> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        try {
            super.onCreate(savedInstanceState)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.adventure, menu)
        return true
    }

    @Throws(IOException::class)
    override fun storeAdventureSpecificValuesInFile(bw: BufferedWriter) {

        bw.write("currentCrewStrength=" + currentCrewStrength + "\n")
        bw.write("currentCrewStrike=" + currentCrewStrike + "\n")
        bw.write("initialCrewStrength=" + initialCrewStrength + "\n")
        bw.write("initialCrewStrike=" + initialCrewStrike + "\n")
        bw.write("log=" + log + "\n")
        bw.write("gold=" + gold + "\n")
        bw.write("booty=" + Adventure.Companion.arrayToString(booty) + "\n")
    }

    override fun loadAdventureSpecificValuesFromFile() {

        currentCrewStrength = Integer.valueOf(savedGame.getProperty("currentCrewStrength"))
        currentCrewStrike = Integer.valueOf(savedGame.getProperty("currentCrewStrike"))
        initialCrewStrength = Integer.valueOf(savedGame.getProperty("initialCrewStrength"))
        initialCrewStrike = Integer.valueOf(savedGame.getProperty("initialCrewStrike"))
        log = Integer.valueOf(savedGame.getProperty("log"))
        gold = Integer.valueOf(savedGame.getProperty("gold"))
        booty = stringToArray(savedGame.getProperty("booty"))
    }

    companion object {

        protected val FRAGMENT_SHIP_COMBAT = 2
        protected val FRAGMENT_EQUIPMENT = 3
        protected val FRAGMENT_BOOTY = 4
        protected val FRAGMENT_NOTES = 5
    }
}
